src/app/SharpRaven/Utilities/RuntimeInfoHelper.cs

```diff
@@ -0,0 +1,100 @@
+            // e.g: .NET Framework 4.7.2633.0, .NET Native, WebAssembly
+            var version = RuntimeInformation.FrameworkDescription;
+#elif NET45
+            var mono = GetFromMono();
+            var version = mono
+                // Windows only: Get latest installation of .NET Framework 4.5 or later
+                ?? Get45PlusLatestInstallationFromRegistry();
+#else
+            var mono = GetFromMono();
+            var version = mono
+                // Environment.Version: NET Framework 4, 4.5, 4.5.1, 4.5.2 = 4.0.30319.xxxxx
+                // .NET Framework 4.6, 4.6.1, 4.6.2, 4.7, 4.7.1 =  4.0.30319.42000
+                // Not recommended on NET45+
+                ?? $".NET Framework {Environment.Version}";
+#endif
+            return version;
+        }
+
+        private static string GetFromMono()
+        {
+            string monoVersion = null;
+
+            var type = Type.GetType("Mono.Runtime", false);
+            if (type != null)
+            {
+                var displayName = type.GetMethod("GetDisplayName", BindingFlags.NonPublic | BindingFlags.Static);
+                if (displayName != null)
+                {
+                    monoVersion = displayName.Invoke(null, null) as string;
+                    if (monoVersion != null)
+                    {
+                        // The implementation of Mono to RuntimeInformation:
+                        // https://github.com/mono/mono/blob/90b49aa3aebb594e0409341f9dca63b74f9df52e/mcs/class/corlib/System.Runtime.InteropServices.RuntimeInformation/RuntimeInformation.cs
+                        // e.g; Mono 5.10.0 (Visual Studio built mono)
+                        monoVersion = "Mono " + monoVersion;
+                    }
+                }
+            }
+
+            return monoVersion;
+        }
+
+        // Only Windows, .NET Framework 4.5 forward
+#if NET45
+        // https://docs.microsoft.com/en-us/dotnet/framework/migration-guide/how-to-determine-which-versions-are-installed#to-find-net-framework-versions-by-querying-the-registry-in-code-net-framework-45-and-later
+        private static string Get45PlusLatestInstallationFromRegistry()
+        {
+            const string subkey = @"SOFTWARE\Microsoft\NET Framework Setup\NDP\v4\Full\";
+
+            try
+            {
+                using (var ndpKey = RegistryKey.OpenBaseKey(RegistryHive.LocalMachine, RegistryView.Registry32).OpenSubKey(subkey))
+                {
+                    return int.TryParse(ndpKey?.GetValue("Release")?.ToString(), out var releaseId)
+                        ? $".NET Framework {CheckFor45PlusVersion(releaseId)}"
+                        : null;
+                }
+            }
+            catch (PlatformNotSupportedException)
+            {
+                return null;
+            }
+
+            string CheckFor45PlusVersion(int releaseKey)
```

Comment:
A bit unintuitive that a method called `Check...()` returns `string`.

Reply:
True. I'll make it: Get